bin/apt-install.sh

```diff
@@ -1,22 +1,58 @@
 #!/bin/bash
 
-apt-get update && \
-    apt-get install -q -y wget gnupg && \
-    wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - && \
-    wget -O - https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | apt-key add - && \
-    echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-15 main" >> /etc/apt/sources.list.d/llvm.list && \
-    echo "deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy-15 main" >> /etc/apt/sources.list.d/llvm.list && \
-    echo "deb https://deb.nodesource.com/node_22.x nodistro main" >> /etc/apt/sources.list.d/nodejs.list && \
-    apt-get update && \
-    apt-get install -q -y automake clang-15 clang-format-15 curl file gawk gcc git jq lld-15 llvm-15 make nodejs parallel unzip zip && \
-    npm install -g bun && \
-    npm install -g prettier
+set -e
 
-update-alternatives --install /usr/bin/clang clang /usr/bin/clang-15 100 && \
-    update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-15 100 && \
-    update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-15 100 && \
-    update-alternatives --install /usr/bin/llc llc /usr/bin/llc-15 100 && \
-    update-alternatives --install /usr/bin/llvm-ar llvm-ar /usr/bin/llvm-ar-15 100 && \
-    update-alternatives --install /usr/bin/llvm-config llvm-config /usr/bin/llvm-config-15 100 && \
-    update-alternatives --install /usr/bin/llvm-link llvm-link /usr/bin/llvm-link-15 100 && \
-    update-alternatives --install /usr/bin/wasm-ld wasm-ld /usr/bin/wasm-ld-15 100
\ No newline at end of file
+default_steps=(llvm node)
+LLVM_VERSION=15
+PRIORITY=100
+
+usage() {
+    echo "Usage: $0 step*"
+    echo "  step is one of llvm, node, othertools"
+    echo "  default steps are: ${default_steps[*]}"
+}
+
+if [ $# -eq 0 ]; then
+    steps=("${default_steps[@]}")
+else
+    steps=("$@")
+fi
+
+for step in "${steps[@]}"; do
+    case "$step" in
+        llvm)
+            apt-get update
+            apt-get install -q -y wget gnupg
+            wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+            echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-$LLVM_VERSION main" >> /etc/apt/sources.list.d/llvm.list
+            echo "deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy-$LLVM_VERSION main" >> /etc/apt/sources.list.d/llvm.list
+            apt-get update
+            apt-get install -q -y automake clang-$LLVM_VERSION clang-format-$LLVM_VERSION curl file gawk gcc git jq lld-$LLVM_VERSION llvm-$LLVM_VERSION make parallel unzip zip
```

Comment:
I find naming this step "llvm" confusing since many of the packages installed here are nothing to do with llvm, just other deps we have. The real distinction in the steps is between things that are needed to build the skiplang toolchain and the rest. I'm not 100% sure that the existing code follows that, or if some thing could be delayed.

Reply:
Not sure either, I'll rename it to `skiplang-deps`